client.go

```diff
@@ -1058,76 +1059,91 @@ func ConvertBrokerIdentityToClientConfig(broker identityClient.Identity, clientU
-
 	signingNone := e3dbClients.RandomNonce()
-
-	tokenSeed := e3dbClients.Base64Encode(notePassword[:]) + e3dbClients.Base64Encode(encodingNonce[:]) + e3dbClients.Base64Encode(signingNone[:])
-
-	publicEncryptionKey, privateEncryptionKey := e3dbClients.DeriveCryptoKey(notePassword[:], encodingNonce[:], e3dbClients.AccountDerivationRounds)
-
-	publicSigningKey, _ := e3dbClients.DeriveSigningKey(notePassword[:], signingNone[:], e3dbClients.AccountDerivationRounds)
-
-	noteBody := NoteBody{
+	// Start accumulating the values that will be needed for a third party to
+	// derive keys for reading and decrypting the broker note
+	tokenSeed := base64NotePassword + e3dbClients.Base64Encode(encodingNonce[:]) + e3dbClients.Base64Encode(signingNone[:])
+	// Derive keys for encrypting and signing the broker note
+	publicEncryptionKey, _ := e3dbClients.DeriveCryptoKey(base64NotePasswordBytes, encodingNonce[:], e3dbClients.IdentityDerivationRounds)
+	publicSigningKey, _ := e3dbClients.DeriveSigningKey(base64NotePasswordBytes, signingNone[:], e3dbClients.IdentityDerivationRounds)
+	// Content for the broker note is all the information needed to instantiate
+	// the broker identity for the realm it can broker actions for
+	rawNoteBody := NoteBody{
 		"realmId":   fmt.Sprintf("%d", broker.RealmID),
 		"realmName": broker.RealmName,
 		"client":    string(serializedBrokerCredentialsBytes),
-		"publicKey": e3dbClients.Base64Encode(publicEncryptionKey[:]),
+		"publicKey": e3dbClients.Base64Encode(publicSigningKey[:]),
 	}
 
+	rawBrokerNote := storageClient.Note{
+		Mode:                e3dbClients.DefaultCryptographicMode,
+		RecipientSigningKey: e3dbClients.Base64Encode(publicSigningKey[:]),
+		WriterSigningKey:    c.StorageClient.SigningKeys.Public.Material,
+		WriterEncryptionKey: c.StorageClient.EncryptionKeys.Public.Material,
+		Data:                rawNoteBody,
+		Type:                "Realm Broker Note",
+		MaxViews:            -1,
+		Expires:             false,
+	}
+	// Sign over all the broker note data and the signature material itself
+	privateSigningKeyBytes, err := e3dbClients.Base64Decode(c.StorageClient.SigningKeys.Private.Material)
+	if err != nil {
+		return "", err
+	}
+	notePrivateSigningKey := [e3dbClients.SigningKeySize]byte{}
+	copy(notePrivateSigningKey[:], privateSigningKeyBytes)
+	signingSalt := uuid.New().String()
+	signedNote, err := c.SignNote(rawBrokerNote, &notePrivateSigningKey, signingSalt)
+	if err != nil {
+		return "", err
+	}
+	// Create and asymmetrically encrypt an access key using the writing clients private key
+	// and the brokers public key, whoever gains access to the broker's private key
+	// (such as being able to derive it by possessing the broker note token)
+	// and the public keys for the writing client which are embedded in the note itself
+	// will be able to decrypt this access key and thus the note contents itself
+	// using the mathematical magic of public key cryptography
```

Comment:
@lkwdwrd to remind me of what the actual science behind the magic is so I can wordsmith the above comment to be less tautological  and instead more conceptual